Dockerfile

```
@@ -0,0 +1,36 @@
+FROM python:3.8-slim
+RUN apt update && apt install -y wget git libglib2.0-0 libsm6 libxext6 libxrender-dev
+RUN apt clean && rm -rf /var/lib/apt/lists/*
+# RUN useradd -ms /bin/bash stablediff
+# install conda
+RUN wget \
+    https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+    && mkdir /root/.conda \
+    && bash Miniconda3-latest-Linux-x86_64.sh -b \
+    && rm -f Miniconda3-latest-Linux-x86_64.sh \
+    && echo PATH="/root/miniconda3/bin":$PATH >> .bashrc 
+RUN chmod +x /root/miniconda3/bin/conda
+RUN ln -s /root/miniconda3/bin/conda /usr/local/bin/conda
+RUN conda update -y conda
+# i am using git clone instead during development of this dockerfile
+COPY . /app/ 
```

So as to take advantage of Docker's [build cache](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache), `COPY`s should be towards the bottom of the Dockerfile as possible. In this case, the only required file required before the `RUN conda env create...` layer is the `environment.yml `.

```suggestion
COPY environment.yml . 
```

---

need to check if really only the environment.yml is needed, I think the .yml uses the setup.py to build the local package as well, that is why I thought it's needed as well. will check it out though